spring-boot-samples/spring-boot-sample-web-secure/src/main/java/sample/web/secure/SampleWebSecureApplication.java

```diff
@@ -60,22 +58,36 @@ public static void main(String[] args) throws Exception {
 		new SpringApplicationBuilder(SampleWebSecureApplication.class).run(args);
 	}
 
-	@Configuration
-	@Order(SecurityProperties.ACCESS_OVERRIDE_ORDER)
-	protected static class ApplicationSecurity extends WebSecurityConfigurerAdapter {
+	@EnableWebSecurity
```

Comment:
At the moment the sample is configuring a protected instance of `AuthenticationManagerBuilder` this is typically used if you want distinct `AuthenticationManager` instances per `WebSecurityConfigurerAdapter`. Since the `AuthenticationMnaagerBuilder` is not the global instance, that means that Boot is unnecessarily generating a random username and password. Without `@EnableWebSecurity` that also means the random password is applicable only for the Boot configured `WebSecurityConfigurerAdapter` instances.

To make it global we must use the `@Autowired` approach. To ensure the ordering is correct we must use `@EnableWebSecurity` or one of its variants as described in https://github.com/spring-projects/spring-boot/pull/5052/files#r64070850